src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,7 +30,12 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
+        String[] addressDetails = address.split(", ");
         this.value = address;
+        this.block = new Block(addressDetails[0], isPrivate);
```

Comment:
think it it makes sense in this context to have individual components private...